igvm/hypervisor_preferences.py

```diff
@@ -101,41 +133,58 @@ def __repr__(self):
 
         return '{}({})'.format(type(self).__name__, args)
 
-    def __call__(self, vm, hv):
+    def get_preference(self, vm, hv) -> Union[float, bool]:
         value = hv.dataset_obj[self.attribute]
 
-        return value is not None, value
+        # if there is no value we assume it's a fresh hv
+        if value is None:
+            return 1.
 
+        # normalize the value. this is only valid for "exhaution values" like
+        # cpu_util_pct and iops_avg
+        return 1 - (value / 100)
 
-class HypervisorAttributeValueLimit(object):
+
+class HypervisorAttributeValueLimit(HypervisorPreference):
     """Compare an attribute value of the hypervisor with the given limit
 
     We are also handling None in here assuming that it is not exceeding
     the limit.  This is coincidentally matching with the None comparison
     on Python 2.  Although our rationale is that those hypervisors being
     brand new.
     """
-    def __init__(self, attribute, limit):
-        self.attribute = attribute
-        self.limit = limit
 
-    def __repr__(self):
+    def __init__(self, attribute: str, limit: int) -> None:
+        self.attribute: str = attribute
+        self.limit: int = limit
+
+    def __repr__(self) -> str:
         args = repr(self.attribute) + ', ' + repr(self.limit)
 
         return '{}({})'.format(type(self).__name__, args)
 
-    def __call__(self, vm, hv):
+    def get_preference(self, vm, hv) -> Union[float, bool]:
         value = hv.dataset_obj[self.attribute]
 
-        return value is not None and value > self.limit
+        # if there is no value we assume it's a fresh hv
+        if value is None:
+            return 1.
+
+        if value > self.limit:
+            return False
 
+        # normalize the value. this is only valid for "exhaution values" like
+        # cpu_util_pct and iops_avg
+        return 1 - (value / 100)
```

Comment:
If there is a slight chance that this becomes a integer we should better cast it to float.

Reply:
done